### PR TITLE
Stop serializing HH config data we don't need

### PR DESCRIPTION
**Pull request by gnidan**
## PR description

As raised by @emilyJLin95, this PR fixes a problem where `hre.config` now contains non-serializable data. Since @truffle/from-hardhat needs only a small portion of the data in `hre.config`, this PR takes the approach to pre-process it, whittling it down to just what we need. This prevents serialization issues that may arise.

## Testing instructions

In repo root:
```console
git checkout fix/from-hardhat-bigints
yarn
cd packages/from-hardhat && yarn link && cd -
```

In a new directory for a fresh HH project:
```console
npx hardhat # follow the wizard, the wonderful wizard
npx hardhat compile
yarn link @truffle/from-hardhat
node
```

In that node repl that just started:
```js
const FromHardhat = require("@truffle/from-hardhat");
await FromHardhat.prepareConfig();
```

Observe that it does not break.

Oh, and you'll need the latest Node.js if you want that `await` to work.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if documentation updates are required.

## Breaking changes and new features

- [x] I have added any needed `breaking-change` and `new-feature` labels for the appropriate packages.

### CODE REVIEW

File: packages/from-hardhat/src/api.ts [hunk 1]
```diff
@@ -92,13 +92,33 @@ export class IncompatibleHardhatVersionError extends Error {
...
+      )
+      .map(([networkName, { url }]) => ({
+        [networkName]: { url }
+      }))
+      .reduce((a, b) => ({ ...a, ...b }), {})
+  });
+
   const hardhatConfig = (await askHardhatConsole(
-    `hre.config`,
+    // stringify the function hooray
```
Comment (cds-amal) [inline packages/from-hardhat/src/api.ts:120]:
```suggestion
    // stringify the function. Hooray!
```

**Comment by gnidan** [comment]
cc @alcuadrado in case it was an accident that `hre.config` is no longer serializable

**Review by cds-amal — APPROVED** [review]
Looks good to me!

File: packages/from-hardhat/src/api.ts (hunk 1 above)
Comment (gnidan) [reply packages/from-hardhat/src/api.ts:120]: i stand by my original angst-driven development tone of voice ;)

**Pull request merged**